python/lib/metrics/README.md

```diff
@@ -1,5 +1,273 @@
+needed hardcoding. Anything that may invoke `dmod.metrics` may be able to perform operations such as:
+
+```python
+from pprint import pprint
+import dmod.metrics as metrics
+
+pprint(metrics.get_metric_options())
+```
+
+and see:
+
+```shell
+[{'description': 'The probability that something was detected. Sensitive to '
+                 'hits, but ignores false alarms. Very sensitive to the '
+                 'climatological frequency of the event. Good for rare events.',
+  'identifier': 'probabilityofdetection',
+  'name': 'Probability Of Detection'},
+ {'description': 'The probability that something was falsely reported as '
+                 'happening. Sensitive to false alarms, but ignores misses. '
+                 'Very sensitive to the climatological frequency of the event. '
+                 'Should be used in conjunction with the probability of '
+                 'detection.',
+  'identifier': 'falsealarmratio',
+  'name': 'False Alarm Ratio'},
+ {'description': 'Sensitive to false alarms, but ignores misses. Can be '
+                 "artificially improved by issuing fewer 'yes' forecasts to "
+                 'reduce the number of false alarms. Not often reported for '
+                 'deterministic forecasts, but is an important component of '
+                 'the Relative Operating Characteristic (ROC) used widely for '
+                 'probabilistic forecasts.',
+  'identifier': 'probabilityoffalsedetection',
+  'name': 'Probability Of False Detection'},
+  ...
+```
+
+This provides more than enough information needed to build user interfaces or gather information
+to provide further context to data. Say I receive a value for `"equitablethreatscore"`. What does
+that value indicate? Well, the metadata explains that the `"equitablethreatscore"` may be
+displayed as `"Equitable Threat Score"` and describes `'How well did the forecast "yes" events correspond 
+to the observed "yes" events (accounting for hits due to chance)? Sensitive to hits. Because it 
+penalises both misses and false alarms in the same way, it does not distinguish the source of 
+forecast error.'`
+
+## What is a `Communicator`?
+
+A `dmod.metrics.Communicator` is an event based mechanism for handling data emission events.
+`Communicator`s are stored within `dmod.metrics.CommunicatorGroup`s which may handle more wide scale
+communication operations. An example can be found in the `score` function of 
+`dmod.metrics.ScoringScheme` where each evaluation of a metric is announced:
+
+```python
+        for metric in self.__metrics:  # type: Metric
+            self.__communicators.info(f"Calling {metric.name}", verbosity=Verbosity.LOUD, publish=True)
+            ...
+```
+
+This means that the `info` event will be triggered on each held `Communicator`, but only on those set to handle 
+messages of a verbosity of `LOUD` or greater, and to call the `write` event after doing so.
+
+Say I have three communicators:
```

Comment:
It would be super nice to see how these three example `Communicators` are created / configured.